knimin/lib/qiita_jira_util.py

```diff
@@ -133,6 +129,43 @@ def _create_kl_jira_project(jira_user, jira_template, study_id, title):
     return jira_project
 
 
+def _assess_replicates(prep):
+    """Renames the technical replciates in the prep information
+
+    Parameters
+    ----------
+    prep : pandas.DataFrame
+        The prep information
+
+    Returns
+    -------
+    pandas.DataFrame, list of str
+        The updates pandas dataframe and a list of replicates
+    """
+    counts = prep.sample_name.value_counts()
+    replicates = counts[counts > 1].index.tolist()
+    # Rename the samples by adding the plate id and the well.
+    # Rename the column sample_name
+    prep.rename(columns={'sample_name': 'original_sample_name'},
+                inplace=True)
+    # Create the new sample name column
+    prep['sample_name'] = prep['original_sample_name']
+    # Rename the samples as needed
+    new_sample_ids = []
+    for r in replicates:
+        for i in prep[prep['original_sample_name'] == r].index:
+            sample_id = _format_sample_id(
+                prep.original_sample_name[i], prep.dna_plate_id[i],
+                prep.row[i] - 1, prep.col[i] - 1)
+            prep.loc[i, 'sample_name'] = sample_id
+            new_sample_ids.append(sample_id)
+
+    # It is ensured that the sample_name column contain unique IDs
+    prep.set_index('sample_name', inplace=True, drop=True)
```

Comment:
If the goal is to check that no sample_names are repeated, then you can add the `verify_integrity` argument to set_index, see here:

http://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.set_index.html